src/lxc/lxc_init.c

```diff
@@ -98,55 +98,61 @@ static struct arguments my_args = {
 static void prevent_forking(void)
 {
 	FILE *f;
-	char name[MAXPATHLEN], path[MAXPATHLEN];
-	int ret;
+	int fd = -1;
+	size_t len = 0;
+	char *line = NULL;
+	char path[MAXPATHLEN];
 
 	f = fopen("/proc/self/cgroup", "r");
-	if (!f) {
-		SYSERROR("Failed to open \"/proc/self/cgroup\"");
+	if (!f)
 		return;
-	}
 
-	while (!feof(f)) {
-		int fd, i;
+	while (getline(&line, &len, f) != -1) {
+		int ret;
+		char *p, *p2;
 
-		if (1 != fscanf(f, "%*d:%" QUOTEVAL(MAXPATHLEN) "s", name)) {
-			ERROR("Failed to parse \"/proc/self/cgroup\"");
-			goto out;
-		}
-		path[0] = 0;
+		p = strchr(line, ':');
+		if (!p)
+			continue;
+		p++;
+		p2 = strchr(p, ':');
+		if (!p2)
+			continue;
+		*p2 = '\0';
 
-		for (i = 0; i < sizeof(name); i++) {
-			if (name[i] == ':') {
-				name[i] = 0;
-				strncpy(path, name + i + 1, sizeof(path));
-				break;
-			}
-		}
+		/* This is a cgroup v2 entry. Skip it. */
+		if ((p2 - p) == 0)
+			continue;
 
-		if (strcmp(name, "pids"))
+		if (strncmp(p, "pids", 5) != 0)
 			continue;
+		p2++;
```

Comment:
can contain newline, needs to be stripped

Reply:
done

Reply:
I...  I hate to be the one breaking it to you...  but I'm afraid you're talking to yourself again.